src/coniql/metrics.py

```diff
@@ -0,0 +1,150 @@
+from typing import Dict, List, Optional
+
+from aioca import Subscription
+from aioca._catools import _Context
+from aiohttp import web
+from aiohttp.hdrs import ACCEPT
+from aiohttp.web_request import Request
+from aioprometheus import (
+    REGISTRY,
+    Counter,
+    Gauge,
+    Summary,
+    count_exceptions,
+    inprogress,
+    timer,
+)
+from aioprometheus.asgi.middleware import EXCLUDE_PATHS
+from aioprometheus.renderer import render
+from graphql import GraphQLError
+from strawberry.aiohttp.handlers import GraphQLTransportWSHandler, GraphQLWSHandler
+from strawberry.extensions import SchemaExtension
+from strawberry.schema import Schema
+from strawberry.subscriptions import GRAPHQL_TRANSPORT_WS_PROTOCOL, GRAPHQL_WS_PROTOCOL
+from strawberry.types import ExecutionContext
+
+# Create all the metrics for the entire application here
+SUBSCRIPTIONS_IN_PROGRESS = Gauge(
+    "coniql_subscriptions_in_progress", "Number of subscriptions in progress"
+)
+REQUEST_TIME = Summary(
+    "coniql_request_processing_seconds", "Time spent processing request"
+)
+REQUESTS = Counter("coniql_request_total", "Total number of requests")
+REQUEST_EXCEPTIONS = Counter(
+    "coniql_request_handler_exceptions",
+    "Number of exceptions and GraphQL errors in requests",
+)
+DROPPED_UPDATES = Gauge(
+    "coniql_dropped_updates", "Number of updates dropped in subscriptions"
+)
+ACTIVE_CHANNELS = Gauge("coniql_active_channels", "Number of active channels in aioca")
+
+
+class MetricsExtension(SchemaExtension):
```

Comment:
Could you review your override class names to make them follow a consistent pattern?

Reply:
Resolved in https://github.com/dls-controls/coniql/pull/53/commits/e9e08e032151a90537479a4a5dcad284a5d23d41